tests/test_dle.py

```diff
@@ -187,13 +187,20 @@ def test_cache_blocking_structure_optrelax():
 
     op = Operator(eqns, opt=('advanced', {'blockrelax': True}))
 
-    bns, _ = assert_blocking(op, {'x0_blk0', 'p_src0_blk0'})
+    bns, _ = assert_blocking(op, {'p_src0_blk0', 'x0_blk0', 'p_src1_blk0'})
```

Comment:
memo for me: I will have to look into this. Why two loop nests over p_src?

Reply:
One for the temp (r0 = src_coords ...) then on for the interpolation. First `p_src` loop is outside the time loop why is before `x0_block`

Reply:
```
  #pragma omp parallel num_threads(nthreads_nonaffine)
  {
    int chunk_size = (int)(fmax(1, (1.0F/3.0F)*(p_sf_M - p_sf_m + 1)/nthreads_nonaffine));
    #pragma omp for schedule(dynamic,chunk_size)
    for (int p_sf = p_sf_m; p_sf <= p_sf_M; p_sf += 1)
    {
      r1[p_sf] = floor(r3*(-o_x + sf_coords[p_sf][0]));
      r2[p_sf] = floor(r4*(-o_y + sf_coords[p_sf][1]));
    }
  }

```

I.e you have those now